### PR TITLE
Added /whatIf operation to deployments at resource group and subscription scope

### PR DESCRIPTION
**Pull request by majastrz**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [X] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/resources/resource-manager/Microsoft.Resources/stable/2019-07-01/resources.json [hunk 1]
```diff
@@ -4983,6 +5170,106 @@
...
+          "description": "The snapshot of the resource before the deployment is executed."
+        },
+        "after": {
+          "readOnly": true,
+          "type": "object",
+          "description": "The predicted snapshot of the resource after the deployment is executed."
+        },
+        "delta": {
+          "readOnly": true,
+          "type": "object",
```
Comment (majastrz) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2019-07-01/resources.json]: >object [](start = 19, length = 6)

I'm intentionally leaving this as an opaque object for now to give us some leeway to alter the format if we discover any issues. This also matches the pattern we followed with the template itself, which is not modeled in the Swagger at all.

Comment (majastrz) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2019-07-01/resources.json]: It turns out we will need to model this after all. I pushed out an update.

---
In reply to: [325962761](https://github.com/Azure/azure-rest-api-specs/pull/7254#discussion_r325962761) [](ancestors = 325962761)

**Review by anthony-c-martin — APPROVED** [review]

**Comment by majastrz** [comment]
@zikalino, can you take a look at this? The pull request is approved from ARM side.

**Comment by majastrz** [comment]
@Azure/azure-sdk-team, @anuchandy, or @zikalino, can one of you take a look and add comments (or approve and merge)? We need to check this in to unblock progress on SDK and then CLI and PowerShell updates.

**Pull request merged**